### ISSUE TITLE
Rename susie plot functions

### ISSUE THREAD

**Issue by gaow** [stephenslab/susieR#27]
Currently we've got `susie_pplot` that plots p-values (convert to -log(p)), PIP or z-scores (convert to -log(p)), and `susie_iterplot` that plots an animation for the VEM fitting process, as you can see [on this page](https://gaow.github.io/mvarbvs/analysis/20180709_SplicingQTL_Detailed.html).

Matthew has commented that we should make the function name `susie_verb_noun`, and in this case `susie_plot_something`. Are there any name suggestions for the `something`? eg. for current `susie_iterplot` change to  `susie_plot_path`? or `susie_plot_fit`? For `susie_pplot`, change to `susie_plot_features`?

**Comment by stephens999** [comment]
I think also rename susie_set_init to susie_init

**Comment by gaow** [comment]
@stephens999 sure! Can i say `susie_plot_iteration` instead? Because one has to have `imagemagick` library installed to get that animation. There might be more native R solutions (package `Animation`) but installation of that thing was a pain. `imagemagick` is a lot easier (i wonder why people write that `Animation` package when other approaches work).


>  And we could
simplify interface by returning univariate z scores, by default, as part of susie object.

Current interface is:

```
Arguments:

    data: can be raw data ‘X’ and ‘y’ as ‘list(X=X,y=y)’, or be a
          vector of z-scores, or p-values.

     res: a susie fit, the output of `susieR::susie()`, or simply the
          posterior inclusion probability matrix alpha.

   dtype: a string indicating the input data type (choices are raw_data
          (for raw data input), p (for p-value input), z (for z-scores
          input) and PIP (for PIP input))

      CS: the output of `susieR::susie_get_CS()`

coverage: coverage of confident sets. Default to 0.9 for 90% confidence
          interval.

 add_bar: add horizontal bar to signals in confidence interval.

       b: for simulated data, specify b = true effects (highlights in
          red).

  max_cs: the biggest CS to display, based on purity (set max_cs in
          between 0 and 1) or size (>1).
```

So to summarize:

1. we make `res` contain what we need and remove `data`? That will involve change in `susie()` function but I guess it is good idea anyways?
2. we change `dtype` to `yaxis`. 
3. Unfortunately we still do need `CS` and `coverage` to make those colored circles, right?

I am always bewildered when programming with R's default data structure. Yes default R data types provides simple starts but it always quickly comes to the point that a customized object makes coding much easier. Unless I do something unconventional (to other R users) I have to stick with `list` or its derivatives. Is that still the 'official' solution in R programming for such situations?

**Comment by pcarbo** [comment]
> Yes default R data types provides simple starts but it always quickly comes to the point that a customized object makes coding much easier. 

@gaow Customized data structures are perfectly fine as long as it is easy for the user to create them. They are used quite widely.

**Comment by stephens999** [comment]
maybe yaxis should be yval ?

I suggest by default we set `CS=NULL` and the plot function automatically
computes the CS from the susie object
(at 0.9 or 0.95 level say). This way the user can simply
susie_plot_CS(fitted)

Matthew

**Comment by gaow** [comment]
> maybe yaxis should be yval ?

Sure

> I suggest by default we set `CS=NULL` 

The problem is `susie_get_CS` does need `Xcorr` to filter CS. So we are back to providing input data `X`. I do not like the idea of adding `Xcorr` to susie object ...

**Comment by stephens999** [comment]
agreed.

one could add CS info to susie object by default. ie after fitting susie,
also compute CS (with reasonable default for level and purity) and add
as an element of fit?
Does computing CS ever take non-trivial computation?

**Comment by gaow** [comment]
> after fitting susie, also compute CS (with reasonable default for level and purity) and add
as an element of fit?

Sure, but we will have to make `susieR::susie()` interface more complicated (with relevant options to compute CS)? Are you sure it is better than complicating the plot interface?

> Does computing CS ever take non-trivial computation?

Not now, not after we used a subset of samples to compute Xcorr.

**Comment by stephens999** [comment]
first reaction is that computing CS is a crucial feature of susie (you
will almost always want to do it after the fit),
and it is just two things (level and purity) for which
we will provide sensible defaults that most users will not need to
change, so it does not really complicate interface.  And it simplifies
susie to basically
a single call. (Users can still compute CS at different levels after
if they want to).

@pcarbo has convinced me that having a single function with many
parameters is not a bad thing if you can provide sensible
defaults for most parameters.

**Comment by gaow** [comment]
> @pcarbo has convinced me that having a single function with many
parameters is not a bad thing if you can provide sensible
defaults for most parameters.

Yes I do see many software packages (particularly methods software eg in `sklearn`) does this, too. Just want to make sure you have no issue with it. I think we are good to make changes.
